realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -124,22 +124,22 @@ public void run() {
                         T instanceToReturn = null;
                         try {
                             instanceToReturn = createRealmOrGetFromCache(configuration, realmClass);
-                        }catch (RuntimeException e) {
+                        } catch (Throwable e) {
                             callback.onError(e);
                         } finally {
-                            fgRealmCreatedLatch.countDown();
+                            createdInForegroundLatch.countDown();
                         }
                         if (instanceToReturn != null) {
                             callback.onSuccess(instanceToReturn);
                         }
                     }
                 });
                 if (!results) {
-                    fgRealmCreatedLatch.countDown();
+                    createdInForegroundLatch.countDown();
                 }
                 // There is a small chance that the posted runnable cannot be executed because of the thread terminated
                 // before the runnable gets fetched from the event queue.
-                if (!fgRealmCreatedLatch.await(5, TimeUnit.SECONDS)) {
+                if (!createdInForegroundLatch.await(5, TimeUnit.SECONDS)) {
```

Comment:
5 seconds will cause an ANR that crashes the app. If we intend to let the app recover from this the value should be less, 1-2 seconds?